src/main/java/org/sagebionetworks/migration/MigrationClientImpl.java

```diff
@@ -38,11 +40,11 @@ public void migrate() {
 	 * Attempt the migration. If there is a failure, retry until the max number of
 	 * retries are exhausted.
 	 */
-	void attemptMigraionWithRetry() {
+	void attemptMigrationWithRetry() {
 		for (int tryCount = 0; tryCount < config.getMaxRetries(); tryCount++) {
 			try {
 				logger.info("Attempting migration try number: " + tryCount + "...");
-				fullMigration.runFullMigration();
+				fullMigration.runFullMigration(actualMaximumNumberOfDestinationJobs(tryCount, config));
```

Comment:
At this level, the MigrationClient only knows the current try number, and should simply pass that along.  It is up to downstream modules to decide how their logic should change based on current try number.